src/encoded/submit.py

```diff
@@ -760,18 +783,252 @@ def create_json_out(self):
             self.json_out['errors'] = self.errors
 
 
+class PedigreeRow:
+
+    def __init__(self, metadata, idx, project, institution):
+        self.project = project
+        self.institution = institution
+        self.row = idx
+        self.metadata = metadata
+        self.errors = []
+        if not self.found_missing_values():
+            self.indiv_alias = generate_individual_alias(project, metadata[SS_INDIVIDUAL_ID])
+            self.individual = self.extract_individual_metadata()
+            self.proband = self.is_proband()
+
+    def found_missing_values(self):
+        # makes sure no required values from spreadsheet are missing
+        missing_required = [col for col in REQUIRED_COLUMNS_PEDIGREE
+                            if col not in self.metadata or not self.metadata[col]]
+        if missing_required:
+            self.errors.append(
+                'Row {} - missing required field(s) {}. This row cannot be processed.'
+                ''.format(self.row, ', '.join(missing_required))
+            )
+        return len(self.errors) > 0
+
+    def format_atid(self, term):
+        """turns HPO or MONDO term IDs into the corresponding @id in the database."""
+        term_id = term.upper().replace('_', ':') if term else ''
+        if re.match(r'^HP:[0-9]{7}$', term_id):
+            return f'/phenotypes/{term_id}/'
+        elif re.match(r'^MONDO:[0-9]{7}$', term_id):
+            return f'/disorders/{term_id}/'
+        else:
+            msg = ('Row {} - term {} does not match the format for an HPO or MONDO '
+                   'ontology term. Please edit and resubmit.'.format(self.row, term))
```

Comment:
Given that `term` can be potentially the empty string, this error message can be confusing. I might suggest using a "repr" style format (which would be "%r" in old-style formatting, and is "!r" in the new style) so that quotes are around it and you can tell when there's nothing there that it's got boundaries. I also prefer the space between the line breaks be at start of line where you can more easily see it's not forgotten than at end of line. And, finally, when you have multiple lines of format string, it may be clearer to use named arguments. So (untested):
```suggestion
            msg = ('Row {row} - term {term!r} does not match the format for an HPO or MONDO'
                   ' ontology term. Please edit and resubmit.'
                   .format(row=self.row, term=term))
```